### ISSUE TITLE
Register field definitions are confusing due to vacuous types

### ISSUE THREAD

**Issue by phausler** [apple/swift-mmio#33]
Register fields are required to specify a type but common usage will use an `as:` parameter. This means that the type that is written for a given field is not really used. Instead it is a type information for storing via an un-inhabited enumeration.

```
@ReadWrite(bits: 0..<3, as: TestValues.self)
public var someField: Test
```

this expands in the register and the field to:
```
public var someField: Test {
    get {
        fatalError()
    }
}

public  enum Test: ContiguousBitField {

            public  typealias Storage = UInt8

            public  static let bitRange = 0 ..< 3
}
```

instead it could be more canonically written in standard swift parlance as:

```
@ReadWrite(bits: 0..<3)
public var someField: TestValues
```

which can expand (by using the name of the field instead of the name of the type)

```
public  enum SomeField: ContiguousBitField {

            public  typealias Storage = UInt8

            public  static let bitRange = 0 ..< 3
}
```

**Comment by fumoboy007** [comment]
I agree! Swapping the projected type and the descriptor type would make it feel more natural.

As I wrote [here](https://forums.swift.org/t/introducing-swift-mmio/68525/23), the register field macro could even have a `generatedDescriptorType` property to allow the developer to specify the generated type name. (It would be nice for the macro to automatically generate a name if the macro property was omitted though.)

```swift
@ReadWrite(bits: 0..<3, generatedTypeName: Test.self)
public var someField: TestValues
```
